tests/integration/test_broadcast.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+
+import cunumeric as num
+from legate.core import LEGATE_MAX_DIM
+
+DIM_CASES = [5, 40]
+
+
+def _broadcast_check(sizes):
+    arr_np = list(np.arange(np.prod(size)).reshape(size) for size in sizes)
+    arr_num = list(num.arange(np.prod(size)).reshape(size) for size in sizes)
+    b = np.broadcast(*arr_np)
+    c = num.broadcast(*arr_num)
+
+    attrs = ["index", "nd", "ndim", "numiter", "shape", "size"]
+
+    is_equal = True
+
+    err_arr = None
+    # test attributes
+    for attr in attrs:
+        if getattr(b, attr) != getattr(c, attr):
+            is_equal = False
+            err_arr = [attr, getattr(b, attr), getattr(c, attr)]
+    if is_equal:
+        # test elements in broadcasted array
+        for each in zip(b, c):
+            if each[0] != each[1]:
+                is_equal = False
+                err_arr = [("iters", b.index), each[0], each[1]]
+                break
+        # test reset method
+        b.reset()
+        c.reset()
+        if b.index != c.index:
+            is_equal = False
+            err_arr = [("reset", b.index), each[0], each[1]]
+        # test whether the broadcast provide views of the original array
+        for i in range(len(arr_num)):
+            arr_num[i][(0,) * arr_num[i].ndim] = 1
+            if c.iters[i][0] != arr_num[i][(0,) * arr_num[i].ndim]:
+                is_equal = False
+                err_arr = [
+                    ("view", i),
+                    c.iters[i][0],
+                    arr_num[i][(0,) * arr_num[i].ndim],
+                ]
+
+    print_msg = f"np.broadcast({sizes})"
+    assert is_equal, (
+        f"Failed, {print_msg}\n"
+        f"Attr, {err_arr[0]}\n"
+        f"numpy result: {err_arr[1]}\n"
+        f"cunumeric_result: {err_arr[2]}\n"
+        f"cunumeric and numpy shows"
+        f" different result\n"
+    )
+
+    print(f"Passed, {print_msg}")
```

Comment:
This check is doing too much. The gold standard for tests is "AAA" &mdash; "Arrange, Act, Assert". Tests should be compact, and test only only one "thing" at at time (ideally with only one assertion, if possible). If tests need to range over different values, that should almost always be done with `pytest.mark.parametrize` decorator, instead of explicit loops. It looks like there are at least three distinct concepts put in here:

* `test_broadcast_attributes` (parametrized by `attrs`)
* `test_broadcast_values` 
* `test_broadcast_views` 

putting everything in one giant check greatly reduces test granularity and means that when there are errors it takes more work to figure out where they actually lie.